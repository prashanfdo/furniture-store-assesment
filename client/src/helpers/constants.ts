export const ROOT_CATEGORY_ID = "156126";
export const DEFAULT_LOCALE = "de_DE";
export const DEFAULT_LOCALE_NUMBER_FORMAT = DEFAULT_LOCALE.replace("_", "-");
