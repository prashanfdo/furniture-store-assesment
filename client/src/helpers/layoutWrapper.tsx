import React from "react";

type LayoutProps = {
  children: React.ReactNode;
};
type ComponentProps = {
  children?: React.ReactNode;
};
const layoutWrapper = (LayoutComponent: React.FC<LayoutProps>) => {
  return (WrappedComponent: React.FC<ComponentProps>) => {
    return class extends React.Component {
      render() {
        return (
          <LayoutComponent>
            <WrappedComponent {...this.props} />
          </LayoutComponent>
        );
      }
    };
  };
};

export default layoutWrapper;
