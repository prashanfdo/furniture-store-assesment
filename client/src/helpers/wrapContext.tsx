import React, { ReactNode } from "react";

const wrapContext = <
  TProviderProps extends { children?: ReactNode },
  TComponentProps extends { children?: ReactNode }
>(
  ContextProvider: React.FC<TProviderProps>,
  WrappedComponent: React.FC<TComponentProps>
) => {
  return (props: TProviderProps & TComponentProps) => {
    return (
      <ContextProvider {...(props as TProviderProps)}>
        <WrappedComponent {...(props as TComponentProps)} />
      </ContextProvider>
    );
  };
};

export default wrapContext;
