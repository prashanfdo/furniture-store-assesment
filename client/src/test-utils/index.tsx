import { render, RenderOptions, renderHook, RenderHookOptions, RenderHookResult } from "@testing-library/react";
import fs from "fs";
import React, { FC, ReactElement } from "react";

import user from "@testing-library/user-event";
import { ProvidersForTests } from "providers";

type GraphQlMocks = React.ComponentProps<typeof ProvidersForTests>["mocks"];

type WrapperProps = React.ComponentProps<typeof ProvidersForTests> & {
  children: React.ReactNode;
};
const Wrapper: FC<WrapperProps> = ({ children, mocks, initialRoute }: WrapperProps) => {
  return (
    <ProvidersForTests mocks={mocks} initialRoute={initialRoute}>
      {children}
    </ProvidersForTests>
  );
};
const wrapper =
  (mocks: GraphQlMocks, initialRoute?: string) =>
  ({ children }: { children: React.ReactNode }) =>
    (
      <Wrapper mocks={mocks} initialRoute={initialRoute}>
        {children}
      </Wrapper>
    );

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper"> & { gqlMocks: GraphQlMocks; initialRoute: string },
) => {
  const view = render(ui, { wrapper: wrapper(options?.gqlMocks, options?.initialRoute), ...options });
  const style = document.createElement("style");
  style.innerHTML = fs.readFileSync("src/index.out.css", "utf8");
  document.head.appendChild(style);

  return view;
};

export const customRenderHook = <T, R>(
  callback: (props: T) => R,
  options?: Omit<RenderHookOptions<T>, "wrapper"> & { gqlMocks?: GraphQlMocks; initialRoute?: string },
): RenderHookResult<R, T> => {
  return renderHook(callback, { ...options, wrapper: wrapper(options?.gqlMocks, options?.initialRoute) });
};

export { faker } from "@faker-js/faker";
export * from "@testing-library/react";
export * from "./generate-data";
export { customRender as render };
export { customRenderHook as renderHook };
export { user };
