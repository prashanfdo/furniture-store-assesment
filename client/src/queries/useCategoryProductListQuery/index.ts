import { useQuery } from "@apollo/client";
import CATEGORY_PRODUCT_LIST_QUERY from "./query";
import { CategoryProductListQueryData, CategoryProductListQueryVars } from "./useCategoryProductListQuery.types";
import { DEFAULT_LOCALE } from "helpers/constants";

export default function useSearchProductListQuery(categoryId: string) {
  return useQuery<CategoryProductListQueryData, CategoryProductListQueryVars>(CATEGORY_PRODUCT_LIST_QUERY, {
    variables: { id: categoryId, locale: DEFAULT_LOCALE },
  });
}
