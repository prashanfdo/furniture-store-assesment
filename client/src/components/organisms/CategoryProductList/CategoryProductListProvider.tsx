import constate from "constate";
import useCategoryProductListQuery from "queries/useCategoryProductListQuery";

type CategoryProductListProviderProps = {
  categoryId: string;
};
const [CategoryProductListProvider, useCategoryProductList] = constate(
  ({ categoryId }: CategoryProductListProviderProps) => {
    const { loading, error, data } = useCategoryProductListQuery(categoryId);
    return {
      loading,
      error,
      data: data?.categories?.[0],
    };
  },
);

export { useCategoryProductList };

export default CategoryProductListProvider;
