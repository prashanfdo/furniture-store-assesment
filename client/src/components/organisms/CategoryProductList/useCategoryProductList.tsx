import useCategoryProductListQuery from "queries/useCategoryProductListQuery";

const useCategoryProductList = (categoryId: string) => {
  const { loading, error, data } = useCategoryProductListQuery(categoryId);
  return {
    loading,
    error,
    data: data?.categories?.[0],
  };
};

export default useCategoryProductList;
