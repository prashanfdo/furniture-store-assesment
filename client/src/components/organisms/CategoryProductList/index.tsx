import { ArticleListHeader } from "components/atoms";
import { ArticleGrid } from "components/molecules";
import useCategoryProductList from "./useCategoryProductList";

type CategoryProductListProps = {
  categoryId: string;
};
const CategoryProductList: React.FC<CategoryProductListProps> = ({ categoryId }: CategoryProductListProps) => {
  const { data, loading, error } = useCategoryProductList(categoryId);

  return (
    <div className="grid grid-rows-[auto_1fr] py-2 px-8">
      {error && <div>Error Occurred. Please try refreshing the page.</div>}
      {(data || loading) && (
        <>
          <ArticleListHeader name={data?.name} articleCount={data?.articleCount} loading={loading} />
          <ArticleGrid products={data?.categoryArticles.articles || []} loading={loading} />
        </>
      )}
    </div>
  );
};

export default CategoryProductList;
