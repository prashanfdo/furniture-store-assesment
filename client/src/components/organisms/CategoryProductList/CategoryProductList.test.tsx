// TODO: clean all below like imports, must import from 'test-utils
import { ROOT_CATEGORY_ID } from "helpers/constants";
import useCategoryProductListQuery from "queries/useCategoryProductListQuery";
import {
  CategoryProductListQueryData,
  CategoryProductListQueryVars,
} from "queries/useCategoryProductListQuery/useCategoryProductListQuery.types";
import { createArticleList, createGqlResult, render } from "test-utils";
import CategoryProductList from "./index";

jest.mock("react-content-loader", () => () => <div data-testid="loading">Loading</div>);
jest.mock("queries/useCategoryProductListQuery");
const mockUseCategoryProductListQuery = useCategoryProductListQuery as jest.MockedFunction<
  typeof useCategoryProductListQuery
>;

describe("CategoryProductList", () => {
  it("should render", () => {
    mockUseCategoryProductListQuery.mockReturnValueOnce(
      createGqlResult<CategoryProductListQueryData, CategoryProductListQueryVars>({
        data: {
          categories: [
            {
              name: "Category 1",
              articleCount: 1,
              categoryArticles: {
                articles: createArticleList(10),
              },
            },
          ],
        },
      }),
    );
    const { container } = render(<CategoryProductList categoryId={ROOT_CATEGORY_ID} />);
    expect(container).toMatchSnapshot();
    expect(mockUseCategoryProductListQuery).toHaveBeenCalledTimes(1);
  });

  it("should render loading state", () => {
    mockUseCategoryProductListQuery.mockReturnValueOnce(
      createGqlResult<CategoryProductListQueryData, CategoryProductListQueryVars>({
        loading: true,
      }),
    );
    const { container } = render(<CategoryProductList categoryId={ROOT_CATEGORY_ID} />);
    expect(container).toMatchSnapshot();
    expect(container.innerHTML).toContain("Loading");
  });

  it("should render error state", () => {
    mockUseCategoryProductListQuery.mockReturnValueOnce(
      createGqlResult<CategoryProductListQueryData, CategoryProductListQueryVars>({
        loading: false,
        error: {
          message: "Test Error",
        } as never,
      }),
    );
    const { container } = render(<CategoryProductList categoryId={ROOT_CATEGORY_ID} />);
    expect(container).toMatchSnapshot();
    expect(container.innerHTML).toContain("Error");
  });
});
