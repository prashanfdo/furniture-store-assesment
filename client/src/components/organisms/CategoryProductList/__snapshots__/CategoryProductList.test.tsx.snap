// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryProductList should render 1`] = `
<div>
  <div
    class="grid grid-rows-[auto_1fr] py-2 px-8"
  >
    <h1
      class="text-3xl grid grid-cols-[auto_auto_1fr] align-middle mt-4 mb-6"
    >
      Category 1
       
      <span
        class="ml-1 text-gray-500"
      >
        (
        1
        )
      </span>
    </h1>
    <div
      class="grid grid-cols-[repeat(auto-fill,_minmax(280px,_1fr))] gap-4"
      data-testid="product-grid"
    >
      <a
        class="block p-4 overflow-hidden bg-white border-none rounded-md decoration-transparent"
        href="/#product-page"
      >
        <div>
          <img
            alt="Tasty Metal Sausages"
            class="object-contain w-full h-96"
            loading="lazy"
            src="https://loremflickr.com/640/480"
          />
        </div>
        <h4
          class="overflow-hidden text-gray-900 text-ellipsis whitespace-nowrap"
        >
          Tasty Metal Sausages
        </h4>
        <div
          class="text-gray-600"
        >
          <span
            class="font-semibold text-red-600"
            data-testid="final-price"
          >
            1.512,00 €
          </span>
        </div>
        <button
          class="text-center border-solid border-gray-200 border block w-full p-2 rounded mt-2 cursor-pointer bg-white text-gray-600 hover:bg-[#7047eb] hover:text-white transition-all duration-100"
        >
          Add To Cart
        </button>
      </a>
      <a
        class="block p-4 overflow-hidden bg-white border-none rounded-md decoration-transparent"
        href="/#product-page"
      >
        <div>
          <img
            alt="Tasty Bronze Bike"
            class="object-contain w-full h-96"
            loading="lazy"
            src="https://loremflickr.com/640/480"
          />
        </div>
        <h4
          class="overflow-hidden text-gray-900 text-ellipsis whitespace-nowrap"
        >
          Tasty Bronze Bike
        </h4>
        <div
          class="text-gray-600"
        >
          <span
            class="font-semibold text-red-600"
            data-testid="final-price"
          >
            2.586,00 €
          </span>
        </div>
        <button
          class="text-center border-solid border-gray-200 border block w-full p-2 rounded mt-2 cursor-pointer bg-white text-gray-600 hover:bg-[#7047eb] hover:text-white transition-all duration-100"
        >
          Add To Cart
        </button>
      </a>
      <a
        class="block p-4 overflow-hidden bg-white border-none rounded-md decoration-transparent"
        href="/#product-page"
      >
        <div>
          <img
            alt="Practical Plastic Chicken"
            class="object-contain w-full h-96"
            loading="lazy"
            src="https://loremflickr.com/640/480"
          />
        </div>
        <h4
          class="overflow-hidden text-gray-900 text-ellipsis whitespace-nowrap"
        >
          Practical Plastic Chicken
        </h4>
        <div
          class="text-gray-600"
        >
          <span
            class="font-semibold text-red-600"
            data-testid="final-price"
          >
            4.743,00 €
          </span>
        </div>
        <button
          class="text-center border-solid border-gray-200 border block w-full p-2 rounded mt-2 cursor-pointer bg-white text-gray-600 hover:bg-[#7047eb] hover:text-white transition-all duration-100"
        >
          Add To Cart
        </button>
      </a>
      <a
        class="block p-4 overflow-hidden bg-white border-none rounded-md decoration-transparent"
        href="/#product-page"
      >
        <div>
          <img
            alt="Luxurious Cotton Gloves"
            class="object-contain w-full h-96"
            loading="lazy"
            src="https://loremflickr.com/640/480"
          />
        </div>
        <h4
          class="overflow-hidden text-gray-900 text-ellipsis whitespace-nowrap"
        >
          Luxurious Cotton Gloves
        </h4>
        <div
          class="text-gray-600"
        >
          <span
            class="font-semibold text-red-600"
            data-testid="final-price"
          >
            2.814,00 €
          </span>
          <span
            class="ml-4 text-sm line-through"
            data-testid="regular-price"
          >
            3.098,00 €
          </span>
          <span
            class="px-1 py-1 ml-2 text-sm text-white bg-red-500 rounded-lg"
            data-testid="discount-percent"
          >
            -
            9
            %
          </span>
        </div>
        <button
          class="text-center border-solid border-gray-200 border block w-full p-2 rounded mt-2 cursor-pointer bg-white text-gray-600 hover:bg-[#7047eb] hover:text-white transition-all duration-100"
        >
          Add To Cart
        </button>
      </a>
      <a
        class="block p-4 overflow-hidden bg-white border-none rounded-md decoration-transparent"
        href="/#product-page"
      >
        <div>
          <img
            alt="Refined Wooden Chair"
            class="object-contain w-full h-96"
            loading="lazy"
            src="https://loremflickr.com/640/480"
          />
        </div>
        <h4
          class="overflow-hidden text-gray-900 text-ellipsis whitespace-nowrap"
        >
          Refined Wooden Chair
        </h4>
        <div
          class="text-gray-600"
        >
          <span
            class="font-semibold text-red-600"
            data-testid="final-price"
          >
            3.682,00 €
          </span>
        </div>
        <button
          class="text-center border-solid border-gray-200 border block w-full p-2 rounded mt-2 cursor-pointer bg-white text-gray-600 hover:bg-[#7047eb] hover:text-white transition-all duration-100"
        >
          Add To Cart
        </button>
      </a>
      <a
        class="block p-4 overflow-hidden bg-white border-none rounded-md decoration-transparent"
        href="/#product-page"
      >
        <div>
          <img
            alt="Modern Cotton Towels"
            class="object-contain w-full h-96"
            loading="lazy"
            src="https://loremflickr.com/640/480"
          />
        </div>
        <h4
          class="overflow-hidden text-gray-900 text-ellipsis whitespace-nowrap"
        >
          Modern Cotton Towels
        </h4>
        <div
          class="text-gray-600"
        >
          <span
            class="font-semibold text-red-600"
            data-testid="final-price"
          >
            615,00 €
          </span>
          <span
            class="ml-4 text-sm line-through"
            data-testid="regular-price"
          >
            1.561,00 €
          </span>
          <span
            class="px-1 py-1 ml-2 text-sm text-white bg-red-500 rounded-lg"
            data-testid="discount-percent"
          >
            -
            61
            %
          </span>
        </div>
        <button
          class="text-center border-solid border-gray-200 border block w-full p-2 rounded mt-2 cursor-pointer bg-white text-gray-600 hover:bg-[#7047eb] hover:text-white transition-all duration-100"
        >
          Add To Cart
        </button>
      </a>
      <a
        class="block p-4 overflow-hidden bg-white border-none rounded-md decoration-transparent"
        href="/#product-page"
      >
        <div>
          <img
            alt="Oriental Soft Fish"
            class="object-contain w-full h-96"
            loading="lazy"
            src="https://loremflickr.com/640/480"
          />
        </div>
        <h4
          class="overflow-hidden text-gray-900 text-ellipsis whitespace-nowrap"
        >
          Oriental Soft Fish
        </h4>
        <div
          class="text-gray-600"
        >
          <span
            class="font-semibold text-red-600"
            data-testid="final-price"
          >
            4.211,00 €
          </span>
        </div>
        <button
          class="text-center border-solid border-gray-200 border block w-full p-2 rounded mt-2 cursor-pointer bg-white text-gray-600 hover:bg-[#7047eb] hover:text-white transition-all duration-100"
        >
          Add To Cart
        </button>
      </a>
      <a
        class="block p-4 overflow-hidden bg-white border-none rounded-md decoration-transparent"
        href="/#product-page"
      >
        <div>
          <img
            alt="Electronic Metal Hat"
            class="object-contain w-full h-96"
            loading="lazy"
            src="https://loremflickr.com/640/480"
          />
        </div>
        <h4
          class="overflow-hidden text-gray-900 text-ellipsis whitespace-nowrap"
        >
          Electronic Metal Hat
        </h4>
        <div
          class="text-gray-600"
        >
          <span
            class="font-semibold text-red-600"
            data-testid="final-price"
          >
            4.460,00 €
          </span>
        </div>
        <button
          class="text-center border-solid border-gray-200 border block w-full p-2 rounded mt-2 cursor-pointer bg-white text-gray-600 hover:bg-[#7047eb] hover:text-white transition-all duration-100"
        >
          Add To Cart
        </button>
      </a>
      <a
        class="block p-4 overflow-hidden bg-white border-none rounded-md decoration-transparent"
        href="/#product-page"
      >
        <div>
          <img
            alt="Licensed Bronze Bacon"
            class="object-contain w-full h-96"
            loading="lazy"
            src="https://loremflickr.com/640/480"
          />
        </div>
        <h4
          class="overflow-hidden text-gray-900 text-ellipsis whitespace-nowrap"
        >
          Licensed Bronze Bacon
        </h4>
        <div
          class="text-gray-600"
        >
          <span
            class="font-semibold text-red-600"
            data-testid="final-price"
          >
            1.839,00 €
          </span>
          <span
            class="ml-4 text-sm line-through"
            data-testid="regular-price"
          >
            2.092,00 €
          </span>
          <span
            class="px-1 py-1 ml-2 text-sm text-white bg-red-500 rounded-lg"
            data-testid="discount-percent"
          >
            -
            12
            %
          </span>
        </div>
        <button
          class="text-center border-solid border-gray-200 border block w-full p-2 rounded mt-2 cursor-pointer bg-white text-gray-600 hover:bg-[#7047eb] hover:text-white transition-all duration-100"
        >
          Add To Cart
        </button>
      </a>
      <a
        class="block p-4 overflow-hidden bg-white border-none rounded-md decoration-transparent"
        href="/#product-page"
      >
        <div>
          <img
            alt="Refined Metal Computer"
            class="object-contain w-full h-96"
            loading="lazy"
            src="https://loremflickr.com/640/480"
          />
        </div>
        <h4
          class="overflow-hidden text-gray-900 text-ellipsis whitespace-nowrap"
        >
          Refined Metal Computer
        </h4>
        <div
          class="text-gray-600"
        >
          <span
            class="font-semibold text-red-600"
            data-testid="final-price"
          >
            2.684,00 €
          </span>
        </div>
        <button
          class="text-center border-solid border-gray-200 border block w-full p-2 rounded mt-2 cursor-pointer bg-white text-gray-600 hover:bg-[#7047eb] hover:text-white transition-all duration-100"
        >
          Add To Cart
        </button>
      </a>
    </div>
  </div>
</div>
`;

exports[`CategoryProductList should render error state 1`] = `
<div>
  <div
    class="grid grid-rows-[auto_1fr] py-2 px-8"
  >
    <div>
      Error Occurred. Please try refreshing the page.
    </div>
  </div>
</div>
`;

exports[`CategoryProductList should render loading state 1`] = `
<div>
  <div
    class="grid grid-rows-[auto_1fr] py-2 px-8"
  >
    <h1
      class="text-3xl grid grid-cols-[auto_auto_1fr] align-middle mt-4 mb-6"
    >
      <svg
        aria-labelledby="v70jvj-aria"
        data-testid="article-list-header-loading"
        height="45"
        role="img"
        width="100%"
      >
        <title
          id="v70jvj-aria"
        >
          Loading...
        </title>
        <rect
          clip-path="url(#v70jvj-diff)"
          height="100%"
          role="presentation"
          style="fill: url(#v70jvj-animated-diff);"
          width="100%"
          x="0"
          y="0"
        />
        <defs>
          <clippath
            id="v70jvj-diff"
          >
            <rect
              height="40"
              rx="5"
              ry="5"
              width="100%"
              x="0"
              y="0"
            />
          </clippath>
          <lineargradient
            id="v70jvj-animated-diff"
          >
            <stop
              offset="0%"
              stop-color="#fff"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="-2; -2; 1"
              />
            </stop>
            <stop
              offset="50%"
              stop-color="#f0f0f0"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="-1; -1; 2"
              />
            </stop>
            <stop
              offset="100%"
              stop-color="#fff"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="0; 0; 3"
              />
            </stop>
          </lineargradient>
        </defs>
      </svg>
    </h1>
    <div
      class="grid grid-cols-[repeat(auto-fill,_minmax(280px,_1fr))] gap-4"
      data-testid="product-grid"
    >
      <svg
        aria-labelledby="8b2unvj-aria"
        data-testid="loading"
        height="300"
        role="img"
        width="100%"
      >
        <title
          id="8b2unvj-aria"
        >
          Loading...
        </title>
        <rect
          clip-path="url(#8b2unvj-diff)"
          height="100%"
          role="presentation"
          style="fill: url(#8b2unvj-animated-diff);"
          width="100%"
          x="0"
          y="0"
        />
        <defs>
          <clippath
            id="8b2unvj-diff"
          >
            <rect
              height="300"
              rx="5"
              ry="5"
              width="100%"
              x="0"
              y="0"
            />
          </clippath>
          <lineargradient
            id="8b2unvj-animated-diff"
          >
            <stop
              offset="0%"
              stop-color="#fff"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="-2; -2; 1"
              />
            </stop>
            <stop
              offset="50%"
              stop-color="#f0f0f0"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="-1; -1; 2"
              />
            </stop>
            <stop
              offset="100%"
              stop-color="#fff"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="0; 0; 3"
              />
            </stop>
          </lineargradient>
        </defs>
      </svg>
      <svg
        aria-labelledby="6zjovkb-aria"
        data-testid="loading"
        height="300"
        role="img"
        width="100%"
      >
        <title
          id="6zjovkb-aria"
        >
          Loading...
        </title>
        <rect
          clip-path="url(#6zjovkb-diff)"
          height="100%"
          role="presentation"
          style="fill: url(#6zjovkb-animated-diff);"
          width="100%"
          x="0"
          y="0"
        />
        <defs>
          <clippath
            id="6zjovkb-diff"
          >
            <rect
              height="300"
              rx="5"
              ry="5"
              width="100%"
              x="0"
              y="0"
            />
          </clippath>
          <lineargradient
            id="6zjovkb-animated-diff"
          >
            <stop
              offset="0%"
              stop-color="#fff"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="-2; -2; 1"
              />
            </stop>
            <stop
              offset="50%"
              stop-color="#f0f0f0"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="-1; -1; 2"
              />
            </stop>
            <stop
              offset="100%"
              stop-color="#fff"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="0; 0; 3"
              />
            </stop>
          </lineargradient>
        </defs>
      </svg>
      <svg
        aria-labelledby="26tzh7-aria"
        data-testid="loading"
        height="300"
        role="img"
        width="100%"
      >
        <title
          id="26tzh7-aria"
        >
          Loading...
        </title>
        <rect
          clip-path="url(#26tzh7-diff)"
          height="100%"
          role="presentation"
          style="fill: url(#26tzh7-animated-diff);"
          width="100%"
          x="0"
          y="0"
        />
        <defs>
          <clippath
            id="26tzh7-diff"
          >
            <rect
              height="300"
              rx="5"
              ry="5"
              width="100%"
              x="0"
              y="0"
            />
          </clippath>
          <lineargradient
            id="26tzh7-animated-diff"
          >
            <stop
              offset="0%"
              stop-color="#fff"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="-2; -2; 1"
              />
            </stop>
            <stop
              offset="50%"
              stop-color="#f0f0f0"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="-1; -1; 2"
              />
            </stop>
            <stop
              offset="100%"
              stop-color="#fff"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="0; 0; 3"
              />
            </stop>
          </lineargradient>
        </defs>
      </svg>
    </div>
  </div>
</div>
`;
